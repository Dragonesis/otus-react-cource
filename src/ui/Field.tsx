import styled from '@emotion/styled'

export interface FieldProps {
  isError?: boolean
}

export const Field = styled.input<FieldProps>`
  transition: border-color 0.2s ease-in-out;
  width: 100%;
  height: 36px;
  border-radius: 4px;
  border: 1px solid #3a3a3a;
  padding-left: 10px;
  padding-right: 10px;
  background-color: #fff;
  border-color: ${({ isError }) => isError && 'red'};

  &:hover,&:focus {
    border-color: #9d9d9d;
  }
`
