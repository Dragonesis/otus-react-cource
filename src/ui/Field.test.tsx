import React from 'react'
import { render } from '@testing-library/react'

import { Field } from './Field'

describe('### Field', () => {
  test('Field renders correctly', () => {
    const { asFragment } = render(<Field />)
    expect(asFragment()).toMatchSnapshot()
  })
})
