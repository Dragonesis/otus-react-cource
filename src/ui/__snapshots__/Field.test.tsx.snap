// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`### Field Field renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-transition: border-color 0.2s ease-in-out;
  transition: border-color 0.2s ease-in-out;
  width: 100%;
  height: 36px;
  border-radius: 4px;
  border: 1px solid #3a3a3a;
  padding-left: 10px;
  padding-right: 10px;
  background-color: #fff;
}

.emotion-0:hover,
.emotion-0:focus {
  border-color: #9d9d9d;
}

<input
    class="emotion-0"
  />
</DocumentFragment>
`;
